jitsi-utils/src/main/java/org/jitsi/utils/config/README.md

```diff
@@ -0,0 +1,133 @@
+
+```
+class WebserverPortProperty extends AbstractConfigProperty<Integer>
+{
+    protected static final configFilePropName = "myapp.webserver.port";
+    protected static final commandLineArgName = "--port";
+
+    public WebserverPortProperty()
+    {
+        // We use the PropertyConfig builder to help define it
+        super(new PropertyConfig<Integer>()
+            // Some function which can get values for command-line args
+            .suppliedBy(() -> getCommandLineValueForArg(commandLineArgName))
+            // Some function which can get values from the config file
+            .suppliedBy(() -> getConfigFileValueForProp(configFilePropName))
+            <...snip: see read frequency and not found strategy sections below>
+        );
+    }
+}
+```
+
+Now we can use this property in the code:
+```
+public class Main
+{
+    private WebserverPortProperty webserverPort = new WebserverPortProperty();
+
+    public static void main(String[] args)
+    {
+        new Webserver(webserverPort.get());
+    }
+}
+```
+
+#### Read frequency strategy
+Some configuration values we want to be able to update if we re-load the config in order to see new values.  To get this behavior, we apply a read frequency strategy to our property definition.  So, if we want the value return from `.get()` to update if the config is reloaded and the value has changed, our property definition would become:
+
+```
+class WebserverPortProperty extends AbstractConfigProperty<Integer>
+{
+    protected static final configFilePropName = "myapp.webserver.port";
+    protected static final commandLineArgName = "--port";
+
+    public WebserverPortProperty()
+    {
+        // We use the PropertyConfig builder to help define it
+        super(new PropertyConfig<Integer>()
+            // Some function which can get values for command-line args
+            .suppliedBy(() -> getCommandLineValueForArg(commandLineArgName))
+            // Some function which can get values from the config file
+            .suppliedBy(() -> getConfigFileValueForProp(configFilePropName))
+            .readEveryTime()
+            <...snip: not found strategy sections below>
+        );
+    }
+}
+```
+And now, the next time we call `.get()` we'll get a new value if the suppliers return a different result than they did before.  To read a property only once (when the property instance is created), use `.readOnce()` instead.
+
+#### Not found strategy
+Generally, configuration properties must be defined (this wrapper defines no method to supply a default value) so if the code looks for a property and it isn't found, a `ConfigPropertyNotFound` exception is thrown.  Sometimes, though, an optional property is desired.  The 'not found' strategy defines what happens when none of the suppliers find a value for this property.  If we want to require a property, we'd use: `.throwIfNotFound()`:
```

Comment:
It might be worth clarifying that it's up to the configuration supplier to provide the default value.

Reply:
would this change address what you're saying? want to make sure i'm understanding you right.

> Generally, configuration properties must be defined (this wrapper defines no method to supply a default value), meaning at least one supplier must provide a value for a given property.  If no suppliers produce a value, a ConfigPropertyNotFound exception is thrown. Sometimes, though, an optional property is desired. The 'not found' strategy defines what happens when none of the suppliers find a value for this property. If we want to require a property, we'd use: .throwIfNotFound()

Reply:
Yeah, that seems good.